src/react/AlertManager.js

```diff
@@ -0,0 +1,270 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2020 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import React, {
+  createContext,
+  useCallback,
+  useEffect,
+  useContext,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { createPortal } from "react-dom";
+import { v4 as uuidv4 } from "uuid";
+import { Alert, Container, Row, Col, Toast } from "react-bootstrap";
+
+const useAlertsContext = ({ timeout } = {}) => {
+  const timersId = useRef([]);
+  const [alerts, setAlerts] = useState([]);
+
+  useEffect(() => {
+    const timersIdRef = timersId.current;
+    return () => timersIdRef.forEach(clearTimeout);
+  }, []);
+
+  const close = useCallback(
+    (alert) => {
+      setAlerts((currentAlerts) => {
```

Comment:
why checking for different array length when you discard the alert anyway?
```
setAlerts((currentAlerts) => {
  if (alert.options.onClose) {
    alert.options.onClose();
  }
  return currentAlerts.filter((a) => a.id !== alert.id);
}
```

Reply:
This hook exposes the alerts array, so you could get a reference on each alert and call `alert.close()` multiple times (by mistake) but you would expect it to be closed just once. `alert.onClose()` should be called only once as well